core/src/main/java/jenkins/fingerprints/FileFingerprintStorage.java

```diff
@@ -216,6 +223,55 @@ public boolean isReady() {
         return new File(Jenkins.get().getRootDir(),"fingerprints").exists();
     }
 
+    /**
+     * Perform Fingerprint cleanup.
+     */
+    public void iterateAndCleanupFingerprints(TaskListener taskListener) {
+        int numFiles = 0;
+
+        File root = new File(getRootDir(), FINGERPRINTS_DIR_NAME);
+        File[] files1 = root.listFiles(f -> f.isDirectory() && f.getName().length()==2);
+        if(files1!=null) {
+            for (File file1 : files1) {
+                File[] files2 = file1.listFiles(f -> f.isDirectory() && f.getName().length()==2);
+                for(File file2 : files2) {
+                    File[] files3 = file2.listFiles(f -> f.isFile() && FINGERPRINT_FILE_PATTERN.matcher(f.getName()).matches());
+                    for(File file3 : files3) {
+                        if(cleanFingerprint(file3, taskListener))
+                            numFiles++;
+                    }
+                    deleteIfEmpty(file2);
+                }
+                deleteIfEmpty(file1);
+            }
+        }
+
+        taskListener.getLogger().println("Cleaned up "+numFiles+" records");
+    }
+
+    private boolean cleanFingerprint(File fingerprintFile, TaskListener listener) {
```

Comment:
If it is private, it does not override the implementation in upper class.  Looks like a 🐛 for API users

Reply:
So this isn't supposed to override the method of the upper class(signature is different). The method in the upper class isn't called by itself, it is -used- by plugin implementations.